src/bip47-test.cpp

```diff
@@ -0,0 +1,273 @@
+                 "xprv9s21ZrQH143K3QTDL4LXw2F7HEK3wJUD2nW2nRk4stbPy6cq3jPPqjiChkVvvNKmPGJxWUtg6LnF5kejMRNNU3TGtRBeJgk33yuGBxrMPHi",
+                 0x80000000)
+                ("xpub68Gmy5EdvgibQVfPdqkBBCHxA5htiqg55crXYuXoQRKfDBFA1WEjWgP6LHhwBZeNK1VTsfTFUHCdrfp1bgwQ9xv5ski8PX9rL2dZXvgGDnw",
+                 "xprv9uHRZZhk6KAJC1avXpDAp4MDc3sQKNxDiPvvkX8Br5ngLNv1TxvUxt4cV1rGL5hj6KCesnDYUhd7oWgT11eZG7XnxHrnYeSvkzY7d2bhkJ7",
+                 1)
+                ("xpub6ASuArnXKPbfEwhqN6e3mwBcDTgzisQN1wXN9BJcM47sSikHjJf3UFHKkNAWbWMiGj7Wf5uMash7SyYq527Hqck2AxYysAA7xmALppuCkwQ",
+                 "xprv9wTYmMFdV23N2TdNG573QoEsfRrWKQgWeibmLntzniatZvR9BmLnvSxqu53Kw1UmYPxLgboyZQaXwTCg8MSY3H2EU4pWcQDnRnrVA1xe8fs",
+                 0x80000002)
+                ("xpub6D4BDPcP2GT577Vvch3R8wDkScZWzQzMMUm3PWbmWvVJrZwQY4VUNgqFJPMM3No2dFDFGTsxxpG5uJh7n7epu4trkrX7x7DogT5Uv6fcLW5",
+                 "xprv9z4pot5VBttmtdRTWfWQmoH1taj2axGVzFqSb8C9xaxKymcFzXBDptWmT7FwuEzG3ryjH4ktypQSAewRiNMjANTtpgP4mLTj34bhnZX7UiM",
+                 2)
+                ("xpub6FHa3pjLCk84BayeJxFW2SP4XRrFd1JYnxeLeU8EqN3vDfZmbqBqaGJAyiLjTAwm6ZLRQUMv1ZACTj37sR62cfN7fe5JnJ7dh8zL4fiyLHV",
+                 "xprvA2JDeKCSNNZky6uBCviVfJSKyQ1mDYahRjijr5idH2WwLsEd4Hsb2Tyh8RfQMuPh7f7RtyzTtdrbdqqsunu5Mm3wDvUAKRHSC34sJ7in334",
+                 1000000000)
+                ("xpub6H1LXWLaKsWFhvm6RVpEL9P4KfRZSW7abD2ttkWP3SSQvnyA8FSVqNTEcYFgJS2UaFcxupHiYkro49S8yGasTvXEYBVPamhGW6cFJodrTHy",
+                 "xprvA41z7zogVVwxVSgdKUHDy1SKmdb533PjDz7J6N6mV6uS3ze1ai8FHa8kmHScGpWmj4WggLyQjgPie1rFSruoUihUZREPSL39UNdE3BBDu76",
+                 0);
+
+void init_test_config() {
+    SoftSetBoolArg("-dandelion", false);
+    ECC_Start();
+    SetupEnvironment();
+    SoftSetBoolArg("-dandelion", false);
+    SetupNetworking();
+    SoftSetBoolArg("-dandelion", false);
+    fPrintToDebugLog = false; // don't want to write to debug.log file
+    fCheckBlockIndex = true;
+    SoftSetBoolArg("-dandelion", false);
+    SelectParams(CBaseChainParams::MAIN);
+    SoftSetBoolArg("-dandelion", false);
+    noui_connect();
+}
+
+
+int main(int argc, char* argv[]) {
+
+    printf("splited test\n");
+
+    init_test_config();
+
+//    std::vector<unsigned char> seed = ParseHex(test1.strHexMaster);
+//    CExtKey key;
+//    CExtPubKey pubkey;
+//    key.SetMaster(&seed[0], seed.size());
+//    pubkey = key.Neuter();
+//    const TestDerivation &derive = test1.vDerive[0];
+//
+//    unsigned char data[74];
+//    key.Encode(data);
+//    pubkey.Encode(data);
+//
+//    // Test private key
+//    CBitcoinExtKey b58key; b58key.SetKey(key);
+//
+//    printf("b58key = %s ,\n derive.prv = %s \n",
+//            b58key.ToString().c_str(),
+//            derive.prv.c_str());
+//
+//    CBitcoinExtKey b58keyDecodeCheck(derive.prv);
+//    CExtKey checkKey = b58keyDecodeCheck.GetKey();
+//    assert(checkKey == key);
+//
+//    printf("Test Passed\n");
+
+//    std::string strHexMaster = "b7b8706d714d9166e66e7ed5b3c61048";
```

Comment:
Any reason why the comment code is there in this file?Better to remove unused code if its not required

Reply:
Hi Akshay! Sorry for the delayed response, it's not super necessary and can be removed

Reply:
I believe this was cleaned up significantly in the following commits.